dapp/src/components/buySell/SwapHomepage.js

```diff
@@ -135,7 +137,7 @@ const SwapHomepage = ({
 
     // currencies flipped
     if (previousSwapMode !== swapMode) {
-      localStorage.setItem(lastSelectedSwapModeKey, swapMode)
+      if (swapMode) localStorage.setItem(lastSelectedSwapModeKey, swapMode)
```

Comment:
Not related to this PR, but this `useEffect` hook is listening to the changes on the `swapMode` var. So, I don't see why there's a need for `previousSwapMode !== swapMode` check here.

Also, `previousSwapMode` and `selectedSwap` aren't being passed to this hook. Unless the `SwapHomepage` component rerenders everytime there is a change in the `ContractStore` state, this may not work as intended

Reply:
Right, removed previousSwapMode. I think the selectedSwap check prevents the component from breaking on initial render so left that there